doc/api/fs.md

```diff
@@ -2884,6 +2894,35 @@ Returns an integer representing the file descriptor.
 For detailed information, see the documentation of the asynchronous version of
 this API: [`fs.open()`][].
 
+## `fs.openWithMkdirp(path[, flags[, mode]], callback)`
```

Comment:
If there's not an elegant way to add this functionality to `open()`, without introducing this new method, I'd rather we start with adding the folder creation option to `fs.write` and `fs.append`.

Reply:
another option is to allow option-bag for 2nd argument: `fs.open(path[, options[, mode]])`.  but yea, being conservative with api is probably best for now.  the main motivation is simplifying scripting-chores with `fs.writeFile`.

Reply:
I think an options bag would be consistent with the other APIs 👍